tasks/initialize.rb

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative '../../ruby_task_helper/files/task_helper.rb'
+require_relative '../lib/cli_helper.rb'
+require 'json'
+require 'open3'
+
+class TerraformInitialize < TaskHelper
+  def init(opts)
+    cli_opts = %w[-no-color]
+    dir = File.expand_path(opts[:dir]) if opts[:dir]
+    cli_opts = cli_opts.join(' ')
+
+    _, _, check_init_status = if dir
+                                CliHelper.execute("terraform providers", dir: dir)
+                              else
+                                CliHelper.execute("terraform providers")
+                              end
+
+    if check_init_status != 0 || opts[:reinit]
```

Comment:
I think the problem (test failures) is that regardless of whether terraform is initialized this status is 0. If you pass the `reinit` arg to the task it should solve it. BUT I think that we need to
1. Re-evaluate how we decide if terraform has been intiialized
2. Do some test cleanup to clean up at the end of each integration test, perhaps something like:
```ruby
  after(:all) do
    terraform_dir = File.join(RSpec.configuration.module_path, '../docker_provision')
    _out, _err, status = Open3.capture3('rm -rf .terraform', chdir: terraform_dir)
    expect(status).to eq(0)
  end
```

I think the reason It was passing locally for you (and me) is that `.terraform` dir was persisting between test runs. I have a super dirty PR with the test fix, it should for sure be refactored https://github.com/puppetlabs/puppetlabs-terraform/pull/13